src/TheThingsNetwork.h

```diff
@@ -58,10 +58,10 @@ class TheThingsNetwork
     TheThingsNetwork(Stream& modemStream, Stream& debugStream, ttn_fp_t fp, int sf = TTN_DEFAULT_SF, int fsb = TTN_DEFAULT_FSB);
     void showStatus();
     void onMessage(void (*cb)(const byte* payload, int length, int port));
-    bool provision(const byte appEui[8], const byte appKey[16]);
-    bool join(const byte appEui[8], const byte appKey[16], int retries = -1, long int retryDelay = 10000);
+    bool provision(const String appEui, const String appKey);
```

Comment:
Tweetonig uses `const char* ` instead of `const String`... Strings are bad right?